application/models/post.go

```diff
@@ -549,3 +549,46 @@ func (p *Post) SetOrigin(location Location) error {
 	p.OriginID = nulls.NewInt(location.ID)
 	return DB.Update(p)
 }
+
+// IsEditable response with true if the given user is the owner of the post or an admin,
+// and it is not in a locked status.
+func (p *Post) IsEditable(user User) (bool, error) {
+	if user.ID <= 0 {
+		return false, errors.New("user.ID must be a valid primary key")
+	}
+
+	if p.CreatedByID <= 0 {
+		if err := DB.Reload(p); err != nil {
+			return false, err
+		}
+	}
+
+	if user.ID != p.CreatedByID && !user.CanEditAllPosts() {
+		return false, nil
+	}
+
+	return p.isStatusEditable(p.Status), nil
+}
+
+// isStatusEditable defines which posts statuses can be edited. It does not use the receiver `p'.
+func (p *Post) isStatusEditable(status string) bool {
+	switch status {
+	case PostStatusOpen:
+		fallthrough
```

Comment:
wow, really?!  This is gross, sorry you have to do switches like this in Go :-D

Reply:
Are you referring to the `fallthrough` keyword? I seem to use `break` more in traditional languages that I use `fallthrough` in Go. I like the change.